packages/salesforcedx-vscode-core/src/messages/i18n.ts

```diff
@@ -155,5 +155,13 @@ export const messages = {
   error_creating_packagexml: 'Error creating package.xml. %s',
   error_extracting_org_source: 'Error extracting downloaded Apex source. %s',
   error_extracting_packages: 'Error extracting packages: %s',
-  error_updating_sfdx_project: 'Error updating sfdx-project.json: %s'
+  error_updating_sfdx_project: 'Error updating sfdx-project.json: %s',
+
+  demo_mode_status_text: `$(gist-secret) SFDX DEMO`,
+  demo_mode_status_tooltip:
+    'You are running the SFDX VS Code extensions in demo mode. You will be prompted if you are connecting to production orgs.',
+  demo_mode_prompt:
+    'Authenticating against a business or production org is not recommended on a demo or shared machine. Continue authenticating or cancel?',
```

Comment:
Can we control the button labels? If so, can we use "Authorize Org" and "Cancel" instead of "Continue" and "Cancel"? We try to make the button labels as specific as we can, especially with dangerous actions, because many users won't read the confirmation message.

My suggestion for this message (unless there's a better way to recommend logging out):

'Authorizing a business or production org is not recommended on a demo or shared machine. If you continue with the authentication, be sure to run "sfdx force:auth:logout --targetusername <your username> --noprompt" when you're done using the org. Are you sure you want to authorize this org?'

Reply:
@ruthemmanuelle Yes, I can control the labels. I can use what you suggested for the button labels.

The resultant message is a bit long but it should be fine since this is not a message that they see normally. So, fortunately, this new notification UI for VS Code should handle it. It would have been a big readability issue with the old notification scheme.

![screen shot 2018-03-06 at 4 07 58 pm](https://user-images.githubusercontent.com/54414/37066105-acb225cc-2158-11e8-9291-cd22afa48ffc.png)

Reply:
After I posted my comment I saw the VS Code command that you added to log out of all orgs, so I edited my comment to use the VS Code version. You're too quick on the draw for me. : )

I think that with the more descriptive button label we can probably eliminate the last sentence, to shorten up the message. What do you think of this?

'Authorizing a business or production org is not recommended on a demo or shared machine. If you continue with the authentication, be sure to run "SFDX: Log Out from All Authorized Orgs" when you're done using this org.'

Reply:
@vazexqi - @mentioning you to make sure you see this, since this comment doesn't show up in the main feed after your most recent commit.

Reply:
![screen shot 2018-03-06 at 4 22 10 pm](https://user-images.githubusercontent.com/54414/37066495-8e16427c-215a-11e8-82f7-ef6d65a66dde.png)

@ruthemmanuelle - Sounds good to me. I have to lift the 'SFDX: Log Out from All Authorized Orgs' up to a top-level command, but that should be fine. This command now shows up even if you are not in demo mode since I think it provides value for non-demo scenarios.